test/powershell/Language/Scripting/MyInvocation.Tests.ps1

```diff
@@ -0,0 +1,49 @@
+﻿Describe 'Testing of MyInvocation' -Tags "CI" {
+
+
+    It 'Script file' {
+
+        $MyInvocation.Line.ToLower().IndexOf('$null = & $scriptblock @parameters') | Should BeGreaterThan -1
+    }
+
+    It 'Function' {
+
+        function myfunc
+        {
+            $MyInvocation.Line.IndexOf("myfunc") | Should BeGreaterThan -1
+        }
+
+        { . myfunc } | Should Not Throw
+        {& myfunc } | Should Not Throw
+    }
+
+    It 'Filter' {
+
+        filter myfilter
+        {
+            $MyInvocation.Line.IndexOf("myfilter") | Should BeGreaterThan -1
+        }
+
+        {. myfilter } | Should Not Throw
+        { & myfilter } | Should Not Throw
+    }
+
+    Context 'Script block' {
+
+        $a = . {$MyInvocation.Line}
+
+        It 'Script block1' { $a.IndexOf('$a = . {$MyInvocation.Line}') | Should BeGreaterThan -1 }
```

Comment:
fixed

---
In reply to: [76685111](https://github.com/Powershell/Powershell/pull/2118#discussion_r76685111) [](ancestors = 76685111)